buildlib/dockers/conda.Dockerfile

```diff
@@ -0,0 +1,109 @@
+ARG CUDA_VERSION=11.2
+ARG UBUNTU_VERSION=20.04
+FROM gpuci/miniconda-cuda:${CUDA_VERSION}-devel-ubuntu${UBUNTU_VERSION}
```

Comment:
can we use some of the docker images in pr/main.yml as baseline?

Reply:
We could, but that would substantially increase complexity, as we have to handle installing and setting up Miniconda and all its dependencies as well. The [gpuCI Dockerfile is publicly available](https://github.com/rapidsai/gpuci-build-environment/blob/branch-21.10/miniconda-cuda/Dockerfile) and maintained by the RAPIDS devops team. IMO, it's best to use that for Python instead of maintaining it ourselves here since they already do that job anyway and have much better experience than me alone to do it.

Reply:
ok, so next question is if we can minimize&refresh the things we add on top of miniconda docker, namely:
- use cuda 11.4
- rdma-core instead of MOFED 5.0
- do we really need to set all these PATH and LD_LIBRARY_PATH below?
- do we really need to create "ucxpy" dir?

Reply:
> * use cuda 11.4
CUDA 11.4 is not yet supported by many packages in conda-forge, some people in the RAPIDS team are working on it together with non-RAPIDS package maintainers, the plan is to have everything there for the next RAPIDS release in October, so before that we can't move to 11.4 yet.

> * rdma-core instead of MOFED 5.0

Why use rdma-core instead of MOFED 5.0? Generally we only test against MOFED today because it's what all our customers have. We could definitely test both, but it's important to have MOFED primarily.

> * do we really need to set all these PATH and LD_LIBRARY_PATH below?

To be honest, I'm not entirely sure, this is based off of https://github.com/openucx/ucx/blob/a994369fbe47df8a1344d825d4ae874e062295ef/buildlib/dockers/ubuntu-release.Dockerfile#L43-L46 .

> * do we really need to create "ucxpy" dir?

No, removed in https://github.com/openucx/ucx/pull/7412/commits/0f99e086b1bfb808c588ecc46530b4934b4024de .